### PR TITLE
Make FORD more flexible to shorten preview deploy time.

### LINKED ISSUE #90: Speed up deployment
The deployment to GitHub pages is quite slow. It seems to be due principally to the transfer of the material.

### PR DESCRIPTION
**Pull request by ccarouge**
# CABLE

Thank you for submitting a pull request to the CABLE Project.

## Description

The deployment of the site takes a long time for the PRs previews. FORD by default builds everything for CABLE trunk and POP code. This resulted in each preview containing about 3000 files. The deployment step needs to deploy everything: the newly built pages and all other existing previews. This means the time to deployment increases very quickly with more open PRs.

This work makes the FORD build more flexible: 

- it only runs if a file has changed
- it only builds the HTML for the modified files

The resulting built previews contain about 100 files for a change in a single file. This means the increase in time to deployment does not dramatically increase with each opened PR. Since the other open pull requests make it impossible to test in the current repository, the behaviour has been tested in a personal test repository. The process for testing was:

- fork the repository: https://github.com/ccarouge/CABLE.
- delete all existing PRs on the fork and the corresponding previews on `gh-pages` branch
- implement the modified action
- open test PRs with modifications to the FORD documentation and monitor the time required for the site to be published.

In the tests, the "Build and deploy PR preview" actions took about 1 min to complete while the `pages build and deployment` action took 1-2 minutes. With the active version of the action in the current repository, the completion times are about 1min30s and 2-4 minutes respectively. The revised action is especially good at keeping the time for the `pages build and deployment` independent of the number of PRs opened, unlike the current version.

The effect will only be visible once all pull requests include this change once it is merged in the `main` branch. Currently opened pull requests can merge the `main` branch if required. But it might be just as fast to finish and merge each pull request instead. 

Fixes #90

## Checklist

- [X] I have checked my code/text and corrected any misspellings

Please add a reviewer when ready for review.

### CODE REVIEW

**Comment by ccarouge** [comment]
@rkutteh I know you don't know much about the automatic checks but if you could have a look and at least see if you can make sense of the changes using the comments etc., that would be good.

**Comment by rkutteh** [comment]
@ccarouge 
- I know I might be a bit pedantic here but I think it is good to specify at the outset that you are talking only about the part of the documentation: "CABLE documentation -- > Developer Guide -- > API documentation" just to be clear that there are no issues coming from the other parts of the documentation (as far as I understand it).  
- Having said that, I still find the comments you made confusing (I also had a look at #90 ) probably because of terminology. Best I ask about it all on zoom because otherwise I will be writing paragraphs...

File: .github/actions/preview/FORD_exclude_files.sh [hunk 1]
```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+# This script excludes all the Fortran files from FORD except these that have
```
Comment (rkutteh) [inline .github/actions/preview/FORD_exclude_files.sh:2]:
```suggestion
# This script excludes all the Fortran files from FORD except those that have
```

File: .github/actions/preview/FORD_exclude_files.sh [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+if [ $proj == "pop" ]
+then
+    echo "Looking for modified files in POP source src_pop/"
+    all_conf_file="pop_ford.md"
+    conf_file="pop_config/pop_config_ford.md"
+    desc_file="pop_config/pop_desc_ford.md"
+    src_dir="src_pop"
+fi
+
+# List of all source files in the trunk
```
Comment (rkutteh) [inline .github/actions/preview/FORD_exclude_files.sh:49]:
```suggestion
# List of the basenames all source file pathnames in the trunk
```

**Comment by ccarouge** [comment]
@rkutteh I have added some explanations in the description at the top of the tests I have performed and added your suggestions.

**Comment by ccarouge** [comment]
We have moved to using Readthedocs and it seems it handles parallel versions much better. We will monitor and reassess as needed. I keep this open for the moment in case it is needed but it could be closed and the branch deleted if it is judged to be unnecessary.

**Comment by ccarouge** [comment]
This is not needed anymore with readthedocs. Closing this PR.

**Pull request closed without merging**